docs/data/toolpad/core/components/app-provider/app-provider.md

```diff
@@ -6,20 +6,90 @@ components: AppProvider
+
+In the following example, an `AppProvider` component wrapping the page provides it with a default theme, and a `DashboardLayout` placed inside it gets its navigation and routing features automatically set based on the props passed to the `AppProvider`.
 
 {{"demo": "AppProviderBasic.js", "height": 500, "iframe": true}}
 
 ## Next.js
 
-The `AppProvider` for Next.js applications includes routing out-of-the-box.
+The `AppProvider` for Next.js applications includes some Next.js integrations out-of-the-box.
+
+By using the specific `AppProvider` for Next.js you do not have to manually configure the integration between some Toolpad features and the corresponding Next.js features (such as routing), making the integration automatic and seamless.
 
 ```tsx
 import { AppProvider } from '@toolpad/core/nextjs/AppProvider';
 // or
 import { AppProvider } from '@toolpad/core/nextjs';
 ```
+
+### Next.js App Router
+
+When using the **Next.js App Router**, the most typical file where to import and use `AppProvider` will be at the top level `layout.tsx` file that defines the layout for all the application pages.
+
+```tsx
+// app/layout.tsx
+
+import { AppProvider } from '@toolpad/core/nextjs/AppProvider';
+
+export default function Layout(props) {
+  const { children } = props;
+
+  return (
+    <html lang="en">
+      <body>
+        <AppProvider>{children}</AppProvider>
+      </body>
+    </html>
+  );
+}
+```
+
+### Next.js Pages Router
+
+When using the **Next.js Pages Router**, the most typical file where to import and use `AppProvider` in order to wrap every page in the application will be the `pages/_app.tsx` file.
+
+```tsx
+// pages/_app.tsx
+
+import { AppProvider } from '@toolpad/core/nextjs/AppProvider';
+
+export default function App(props) {
+  const { Component, pageProps } = props;
+
+  return (
+    <AppProvider>
+      <Component {...pageProps} />
+    </AppProvider>
+  );
+}
+```
+
+## Theming
+
+An `AppProvider` can set a visual theme for all elements inside it to adopt via the `theme` prop. This prop can be set in a few distinct ways with different advantages and disadvantages:
+
+1. [CSS variables theme](https://mui.com/material-ui/experimental-api/css-theme-variables/overview/): the default and recommended theming option for Toolpad applications, as it is the only option that prevents issues such as [dark-mode SSR flickering](https://github.com/mui/material-ui/issues/27651) and supports both light and dark mode with a single theme definition. The provided default theme in Toolpad is already in this format.
+2. [Standard Material UI theme](https://mui.com/material-ui/customization/theming/): a single standard Material UI theme can be provided as the only theme to be used.
+3. **Light and dark themes**: two separate Material UI themes can be provided for light and dark mode in an object with the format `{ light: Theme, dark: Theme }`
+
+{{"demo": "AppProviderTheme.js", "height": 500, "iframe": true}}
+
+### Predefined themes
```

Comment:
I'm not 100% sure about this. Isn't the predefined theme just the default MUI theme? I'm not sure providing different themes should be in the scope of `@toolpad/core`. Or is there a specific reason?

In any case, I think we need to exclusively promote css vars themes and leave the legacy functionality for when people bring their own existing theme.

Reply:
Doesn't necessarily need to be in the scope, I just wanted to possibly provide some ready-to-use themes that will look good. And that default theme just has a few color changes to look a bit better, but I guess the default theme should look decent enough.

To keep things simple let's keep themes outside the library then, and will rechecked how the default MUI theme looks - if it looks good we can keep it as the default.